src/expressions/functions/builtInFunctions_fontoxpath.ts

```diff
@@ -69,64 +53,41 @@ function buildResultIterator(
 	const contextItemSequence = variables['.'] ? variables['.']() : sequenceFactory.empty();
 	delete variables['.'];
 
-	const executionSpecificStaticContext = new ExecutionSpecificStaticContext(
-		(prefix) => staticContext.resolveNamespace(prefix),
-		Object.keys(variables).reduce((vars: { [s: string]: string }, varName) => {
-			vars[varName] = varName;
-			return vars;
-		}, {}),
-		BUILT_IN_NAMESPACE_URIS.FUNCTIONS_NAMESPACE_URI,
-		(lexicalName, arity) => staticContext.resolveFunctionName(lexicalName, arity)
-	);
-	const innerStaticContext = new StaticContext(executionSpecificStaticContext);
-
-	const ast = createAstFromValue(queryValue, executionParameters.debug);
-
-	const prolog = astHelper.followPath(ast, ['mainModule', 'prolog']);
-	if (prolog) {
-		processProlog(prolog, innerStaticContext);
-	}
-
-	annotateAst(ast, new AnnotationContext(innerStaticContext));
-
-	const queryBodyContents = astHelper.followPath(ast, ['mainModule', 'queryBody', '*']);
-
-	const selector = compileAstToExpression(queryBodyContents, {
-		allowUpdating: false,
-		allowXQuery: true,
-	});
-
 	try {
-		selector.performStaticEvaluation(innerStaticContext);
-	} catch (error) {
-		printAndRethrowError(queryValue.value, error);
-	}
-
-	const variableBindings = Object.keys(variables).reduce((variablesByBindingKey, varName) => {
-		variablesByBindingKey[executionSpecificStaticContext.lookupVariable(null, varName)] =
-			variables[varName];
-		return variablesByBindingKey;
-	}, Object.create(null));
-
-	const context = contextItemSequence.isEmpty()
-		? {
-				contextItem: null,
-				contextItemIndex: -1,
-				contextSequence: contextItemSequence,
-				variableBindings,
-		  }
-		: {
-				contextItem: contextItemSequence.first(),
-				contextItemIndex: 0,
-				contextSequence: contextItemSequence,
-				variableBindings,
-		  };
-
-	const innerDynamicContext = new DynamicContext(context);
+		const { expression, staticContext: innerStaticContext } = staticallyCompileXPath(
+			getEvaluableExpressionFromValue(queryValue),
+			{
+				allowUpdating: false,
+				allowXQuery: true,
+				debug: executionParameters.debug,
+				// TODO: should we inherit this from the outer evaluation somehow?
+				disableCache: false,
```

Comment:
It might be good to disable this cache if the cache was disabled for the outer evaluation, but it doesn't seem like information this is currently passed along anywhere. Should I add this property to executionParameters?

Reply:
Cool, sounds great